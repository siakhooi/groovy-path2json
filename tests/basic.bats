setup() {
    load 'common-setup'
    common_setup

    outputExpected=$BATS_TEST_FILENAME.$BATS_TEST_DESCRIPTION.expected.json
}

@test "basic" {
    tempDir=$(mktemp -d)
    cd "$tempDir"
    run path2json

    sed "s:TEMPDIR:${tempDir}:" "$outputExpected" | assert_output
}
@test "subdirectories" {
    tempDir=$(mktemp -d)
    cd "$tempDir"
    mkdir -p a/b/c
    echo -n '1234' > a/b/c/d
    echo -n '123' > a/b/e
    run path2json

    sed "s:TEMPDIR:${tempDir}:g" "$outputExpected" | assert_output
}
