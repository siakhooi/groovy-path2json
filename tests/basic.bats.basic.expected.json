{"creationTime":"2024-12-10T12:06:41.273863317Z","creationTimeInMillis":1733832401273,"lastAccessTime":"2024-12-10T12:06:41.273863317Z","lastAccessTimeInMillis":1733832401273,"lastModifiedTimeInMillis":1733832401273,"isDirectory":true,"isRegularFile":false,"size":4096,"path":"TEMPDIR","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T12:06:41.273863317Z","isOther":false,"children":[]}